Create rule S6660 (#2383)
SonarSource/rspec: rules/S6660/javascript/rule.adoc

== Why is this an issue?

When `if` is the only statement in the `else` block, it is better to use `else if` because it simplifies the code and makes it more readable.

When using nested `if` statements, it can be difficult to keep track of the logic and understand the flow of the code. Using `else if` makes the code more concise and easier to follow.

> [source,javascript]

Reviewer: add diffs
```suggestion
[source,javascript,diff-id=1,diff-type=noncompliant]
```

Reviewer: ask for details if you've never used them

----
if (condition1) {
    // ...
} else {

>     if (condition2) {  // Noncompliant

Reviewer: ```suggestion
    if (condition2) {  // Noncompliant: 'if' statement is the only statement in the 'else' block
```

        // ...
    }
}

if (condition3) {
    // ...
} else {

>     if (condition4) { // Noncompliant

Reviewer: ```suggestion
    if (condition4) { // Noncompliant: 'if' statement is the only statement in the 'else' 
```

        // ...
    } else {
        // ...
    }
}
----

> Fix your code by using `else if` instead of nested `if` block.

Reviewer: ```suggestion
Fix your code by using `else if` if the nested `if` is the only statement.
```

[source,javascript]
----
if (condition1) {
    // ...
} else if (condition2) {
    // ...
}

if (condition3) {
    // ...
} else if (condition4) {
    // ...
} else {
    // ...
}
----

== Resources
=== Documentation

> * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/if\...else[MDN - if...else]

Reviewer: ```suggestion
* link:++https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/if...else++[MDN - ``++if...else++``]
```
